flytekit/types/structured/structured_dataset.py

```diff
@@ -279,14 +299,17 @@ def to_literal(
         # StructuredDataset instance.
         if issubclass(python_type, StructuredDataset):
             assert isinstance(python_val, StructuredDataset)
-            df_type = type(python_val.dataframe)
+            python_val.uri = python_val.uri or ctx.file_access.get_random_remote_path()
```

Comment:
I don't think we should be using `file_access` in this file.  The determination of what the uri should be should be left entirely up to the individual encoders/decoders.  Is there a reason this was needed?

Reply:
if we move to encoders, every encoder will have `file_access` because `python_val.uri` might be empty. People will not have good UX if they want to extend new encodes because they still need to handle empty URI or use file_access in the encoder.